### DISCUSSION TITLE
Is stacking of facts possible?

### DISCUSSION THREAD

**Discussion by angelabriel** [trento-project/wanda#129] [Q&A] [answered]
If using more than one facts gatherer is the execution order of this facts the same as in the check yaml file or will they get sorted in some other way?
Is it possible to stack facts, so that the result of one gatherer can be used as argument for another, following gatherer?
And if yes, can I use  Rhai expressions on the facts used as argument for the follow-up gatherer?

**Comment by fabriziosestito** [comment, marked as answer]
Dependency between gatherers is on the radar but not yet implemented.

As an initial step, for simple dependencies, we plan to implement custom gatherers for the moment that will take care of the argument passing. 
For instance, _sbd_dump_ gatherer could fetch the devices list from the conf file and then use them to use the return value as an object map.

Rhai cannot call fact-gatherers directly, because expectations are run after the fact-gathering phase.
